demos/meltdown_de.cc

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Demonstrates the Meltdown-DE on AMD.
+ * We exploit the fact that the data from the remainder on AMD is speculatively
+ * computed this way (before the #DE exception is raised and the remainder is
+ * zeroed):
+ * 0 % 0 = 0
+ * 1 % 0 = 2
+ * 2 % 0 = 2
+ * 3 % 0 = 3
+ * 4 % 0 = 2
+ * 5 % 0 = 2
+ * 6 % 0 = 3
+ * 7 % 0 = 3
+ * 8 % 0 = 2
+ * 9 % 0 = 2
+ * 10 % 0 = 2
+ * 11 % 0 = 2
+ * 12 % 0 = 3
+ * 13 % 0 = 3
+ * 14 % 0 = 3
+ * 15 % 0 = 3
+ * 16 % 0 = 2
+ * 7 times more 2
```

Comment:
Not sure how to read this comment. Does this mean that `x % 0 == 2` for x \in {17..23}?

Reply:
Yes.

Reply:
Extended the comment.